src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -308,6 +307,26 @@ public ParameterizedClass getTokenRangeSplitter(RepairType repairType)
         return applyOverrides(repairType, opt -> opt.token_range_splitter);
     }
 
+    /**
+     * Set a new token range splitter, this is not meant to be used other than for testing.
+     */
+    public void setTokenRangeSplitter(RepairType repairType, ParameterizedClass tokenRangeSplitter)
+    {
+        getOptions(repairType).token_range_splitter = tokenRangeSplitter;
+        tokenRangeSplitters.remove(repairType);
+    }
+
+    public IAutoRepairTokenRangeSplitter getTokenRangeSplitterInstance(RepairType repairType)
+    {
+        IAutoRepairTokenRangeSplitter splitter = tokenRangeSplitters.get(repairType);
+        if (splitter == null)
+        {
+            splitter = newAutoRepairTokenRangeSplitter(repairType, getTokenRangeSplitter(repairType));
+            tokenRangeSplitters.put(repairType, splitter);
+        }
+        return splitter;
```

Comment:
Prefer `computeIfAbsent`
```suggestion
        return tokenRangeSplitters.computeIfAbsent(repairType,
                                                   key -> newAutoRepairTokenRangeSplitter(key, getTokenRangeSplitter(key)));
```